### PR TITLE
add facebook-like button

### PR DESCRIPTION
**Pull request by P1tt187**
add button for facebook like functionality
click result is a popup with the clickable button

### CODE REVIEW

**Comment by ClemensSahs** [comment]
any progress?

File: src/js/services/facebooklike.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            'en': 'like'
+        },
+        name: 'facebooklike',
+        title: {
+            'de': 'Gefällt mir bei Facebook ',
+            'en': 'Like on Facebook'
+        },
+        shareUrl: 'http://www.facebook.com/plugins/like.php?href=' + fbEncUrl + shariff.getReferrerTrack() + '&width=80&layout=button_count&action=like&show_faces=false&share=false&height=30',
+        popupInfoText: {
+            'de': 'Zum Schutz Ihrer Daten nehmen wir erst jetzt Kontakt mit Facebook auf. Klicken Sie bitte den nachfolgenden Button um diese Seite zu liken.'
```
Comment (pmb0) [inline src/js/services/facebooklike.js]: Das ist eher eine Notlösung, als eine Lösung. Kennt jemand einen Weg, einen Like durch einen einfachen Request zu bewirken, anstatt ein Popup mit dem originalen Like-Button zu öffnen?

**Comment by pmb0** [comment]
Hi!

Thanks for your work. There are a lot of features and changes provided by this pull request: facebook-like, XING, divisor, iframe, ... Some of them are not finished yet ("// possibility to do something like 24.6k shares (not finally implemented yet)").

Facebook likes: I don't like the way the feature is implemented. A popup window opens where the original facebook like button appears. So the use has to click twice. We want a one-click button, don't we?

And the Shariff code base changed, so we need some extra time to merge the changes. Give us some time (or provide smaller, more atomic pull requests).

**Comment by simonberger** [comment]
Hi.

sure we want a one-click version of facebook-like but it seems to not beeing possible. We can load the button from facebook and display it via iframe for example.
But we aren't allowed to read or work with this data, surely an automatic click (was our first idea too) isn't possible out of this reason.
I don't see any chance to implement it in a different way.

In the first version of this pull request it got displayed in a popup, a second try replaced the "custom-button" with the one from facebook (not user friendly / no clear flow).
The newest version adds a tooltip on click with a small description text.

Yes there are some other interesting changes. Xing has been taken from yanniks fork.
The "division changes" are working but indeed are not finally complete. You can change the share count from current 25k to for example 25.123 (same as facebook) and you can step back to 100k if you reach 100 000 (customizable limit). Just the scenario you wrote can't be done yet.

I can give you a productive example/link of the facebook-like tooltip / share-count use if you are interested.

**Comment by pmb0** [comment]
I certainly am interested in an example.

**Comment by simonberger** [comment]
https://shop.tele2.de

At the bottom of the page.

Container start:

<div class="shariff" data-services="[&quot;facebooklike&quot;,&quot;xing&quot;]" data-url="https://www.tele2.de" data-width="130" data-count-division-start="100000">
</div>

File: src/js/services/facebooklike.js (hunk 1 above)
Comment (P1tt187) [inline src/js/services/facebooklike.js]: eine möglichkeit wäre über window.postMessage den request im hintergrund abzuschicken
die antwort erhält man in Form eines events.
Ich weiß allerdings nicht, wie der aufruf für die like-action aussehen muss

**Comment by ipa01** [comment]
Since shariff was explicitly developed to allow sharing with just one click, we will not implement facebook like at the moment.
Nevertheless, it is available in the older two-click-buttons (http://www.heise.de/extras/socialshareprivacy/).

Some selected changes will be merged in the following days.

**Comment by simonberger** [comment]
Out of concept an absolutely understandable decision.
But this feature is in my opinion just more important for many uses than the sharing.
At the first look on this new social sharing plugin you might not note it can "only" share links.

In use and look its so much better than the old heise plugin.

**Comment by compeak** [comment]
I am closing this, because the "Like" button is out of concept.

- Shariff is meant as a tool for sharing (putting information into something). "Like" upvotes content that already exists somewhere.
- "Like" is a special Facebook thing. We do not want to implement backend specific functionality. (We would not implement "[face pile](https://www.google.de/search?q=face+pile&biw=1600&bih=1014&tbm=isch&gbv=2&sei=V7moVqjJLsq3afbDrqAE)" either.)

**Comment by drupalsmitty** [comment]
There is a Facebook-Like-Button Integration now at https://github.com/richard67/shariff-plus

Wouldn't it be useful to rethink the decision about the "Like" button being out of concept and perhaps join the two projects?

**Comment by richard67** [comment]
@drupalsmitty I don’t think maintainers will react on comments on closed pull requests. Maybe you should open a new issue for your question. If the maintainers change their decision about the Facebook like button, I will make a pull request to integrate my changes into Shariff. I am the guy who made the integration you referred to.

**Comment by compeak** [comment]
> I don’t think maintainers will react on comments on closed pull requests.

@richard67 We try to, but there is so little time and so much to do.

> Wouldn't it be useful to rethink the decision about the "Like" button being out of concept and perhaps join the two projects?

@drupalsmitty I'm rather content with the like button being available through a fork and without compromising the concept of shariff.

**Comment by richard67** [comment]
> without compromising the concept of shariff.

@compeak I know that the concept of Shariff is to be used for sharing, but in the Ct' article which introduced Shariff and also in a recent article which promotied it again, the introducing texts mentioned also other buttons. Here the example from [https://www.heise.de/ct/artikel/Shariff-Social-Media-Buttons-mit-Datenschutz-2467514.html](https://www.heise.de/ct/artikel/Shariff-Social-Media-Buttons-mit-Datenschutz-2467514.html):

> Dann können Sie nach Gutdünken liken, +1en oder tweeten – mehr Informationen erhält das soziale Netzwerk nicht.

The other example, that recent article (could have been end of 2017 or beginning of this year), I can't find it right now.

If you really want that users know about the concept of Shariff, then Ct' have to clean up their info from such old text fragments which to me seem to be remainders from the old solution before Shariff, [https://www.heise.de/ct/artikel/2-Klicks-fuer-mehr-Datenschutz-1333879.html](https://www.heise.de/ct/artikel/2-Klicks-fuer-mehr-Datenschutz-1333879.html).

**Pull request closed without merging**